security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -83,10 +89,10 @@ func (authn *K8sSvcAcctAuthn) reviewServiceAccountAtK8sAPIServer(targetToken str
 		Kind:       "TokenReview",
 		Spec: specForSaValidationRequest{
 			Token: targetToken,
```

Comment:
To confirm: for 1.12 we are still subject to 'confused deputy' ? My understanding was that we will check the audience in citadel ( by decoding the token and comparing ) so it works on 1.12 as well. 

If in 1.12 audience check doesn't work and we don't add the fix - it should probably be in release notes, so users avoid using SDS ( or don't use any JWT tokens with any other servers that may create a 'confused deputy' problem)

Reply:
If adding release notes, please do it here: https://github.com/istio/istio/wiki/Release-Note-1.3-Draft

Reply:
I've updated the release notes.

Reply:
Did we agree to completely break SDS for k8s 1.12 in TOC? My understanding is we just agreed to get rid of legacy jwt?

Reply:
Yes, we agreed to get rid of legacy jwt.

Reply:
> Use trustworthy jwts only in 1.3 for SDS (requires k8s 1.13 or above).

From TOC meeting:

> The Trustworthy JWT is supported starting with K8s V1.12. For on-prem clusters, admin need to pass flags to the apiserver to enable it.

Dropping support for 1.12 is much different than for 1.11... GKE still uses 1.12 as the default

Reply:
GKE now has 1.13.7, so I think it's fine to ask customers to use 1.13? Also, Istio claims to support 1.13-1.15 https://preliminary.istio.io/faq/general/#what-deployment-environment for 1.3.

Reply:
I think we are not breaking SDS in K8s 1.12. SDS will still work, it's just K8s doesn't check the JWT "aud" field in 1.12, right? @pitlv2109

Reply:
Nope, it's a breaking change. K8s below 1.13 fails to check if aud is anything other than api server. 
Technically, we could still support 1.12 if in Citadel we parse the jwt and check if the audience is istio-ca, and send the TokenReview without the audience. I think you suggested this but I thought just bump the support to 1.13 and let k8s api does all the work.

Reply:
I'm happy to let Citadel does the dirty work so that we can support k8s 1.12 or encourage users to upgrade to a newer k8s verion for other benefits, including security, that k8s > 1.12 offers.

Reply:
Sorry, I didn't know this is a breaking change to 1.12. Given that SDS is an Alpha feature, I think it's acceptable to ask users to upgrade to 1.13 to use SDS (Also technically this doesn't break our policy of supporting the latest 3 K8s versions). We will update the release notes, blog post and the SDS task.

Having Citadel to check the audience isn't nice IMO. If we decide to check the audience in Citadel, we must not check it on apiserver.

Reply:
Sounds good, so I'll keep things as they are. I've updated all the relevant docs.